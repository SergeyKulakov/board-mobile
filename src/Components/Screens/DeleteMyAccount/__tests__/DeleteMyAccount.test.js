import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import Component from '../DeleteMyAccount'

describe('DeleteMyAccount', () => {
  const defaultProps = {
    activeLanguage: 'en',
    getError: () => null,
    navigate: {
      hideModal: () => null,
    },
    onDeleteAccount: () => null,
    onShowPuck: () => null,
    t: text => text,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    const renderListFooter = wrapper.instance().renderListFooter()
    const renderListHeader = wrapper.instance().renderListHeader()
    const renderRow = wrapper.instance().renderRow({
      item: {
        value: true,
        onClick: () => null,
        text: 'text',
      },
    })

    expect([
      wrapper,
      renderListFooter,
      renderListHeader,
      renderRow,
    ]).toMatchSnapshot()
  })

  it('call handleChangeValue method', () => {
    const wrapper = shallow(<Component {...defaultProps} />)

    wrapper.instance().input = {
      current: {
        handleSetFocus: jest.fn(),
      },
    }

    wrapper.instance().handleChangeReason('value')

    expect(wrapper.state('reason')).toBe('value')

    wrapper.instance().handleChangeValue(2)

    expect(wrapper.state('checkedValue')).toBe(2)
    expect(wrapper.state('reason')).toBe('')
    expect(
      wrapper.instance().input.current.handleSetFocus,
    ).not.toHaveBeenCalled()

    wrapper.instance().handleChangeReason('value2')

    expect(wrapper.state('reason')).toBe('value2')

    wrapper.instance().handleChangeValue(6)

    expect(wrapper.state('checkedValue')).toBe(6)
    expect(wrapper.state('reason')).toBe('')
    expect(
      wrapper.instance().input.current.handleSetFocus,
    ).toHaveBeenCalledTimes(1)
  })

  describe('call submit action', () => {
    it('with warning', () => {
      const props = {
        ...defaultProps,
        onShowPuck: jest.fn(),
        onDeleteAccount: jest.fn(),
        getError: jest.fn(),
      }
      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({ checkedValue: 6 })

      wrapper.instance().handleSubmit()

      expect(props.onDeleteAccount).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()
      expect(props.onShowPuck).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).toHaveBeenCalledWith({
        type: 'warning',
        message: 'deleteAccountScreen.writeYourReason',
      })
    })

    const getProps = isSuccessful => ({
      ...defaultProps,
      onDeleteAccount: jest.fn((request, callback) => {
        setTimeout(() => {
          callback({
            error: isSuccessful
              ? undefined
              : { payload: { code: 'some error code' } },
          })
        }, 1000)
      }),
      onShowPuck: jest.fn(),
      getError: jest.fn(),
    })

    it('successful', done => {
      const props = getProps(true)

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({ checkedValue: 3 })

      wrapper.instance().handleSubmit()

      expect(wrapper.state('isRequest')).toBe(true)
      expect(props.onDeleteAccount).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('isRequest')).toBe(false)
        expect(props.onDeleteAccount).toHaveBeenCalledTimes(1)
        expect(props.onShowPuck).not.toHaveBeenCalled()
        expect(props.getError).not.toHaveBeenCalled()

        done()
      }, 1100)
    })

    it('failure', done => {
      const props = getProps(false)

      const wrapper = shallow(<Component {...props} />)

      wrapper.setState({ checkedValue: 3 })

      wrapper.instance().handleSubmit()

      expect(wrapper.state('isRequest')).toBe(true)
      expect(props.onDeleteAccount).toHaveBeenCalledTimes(1)
      expect(props.onShowPuck).not.toHaveBeenCalled()
      expect(props.getError).not.toHaveBeenCalled()

      setTimeout(() => {
        expect(wrapper.state('isRequest')).toBe(false)
        expect(props.onDeleteAccount).toHaveBeenCalledTimes(1)
        expect(props.onShowPuck).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledTimes(1)
        expect(props.getError).toHaveBeenCalledWith({
          payload: { code: 'some error code' },
        })

        done()
      }, 1100)
    })
  })
})
