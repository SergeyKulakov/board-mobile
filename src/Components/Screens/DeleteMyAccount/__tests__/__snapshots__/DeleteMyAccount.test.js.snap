// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMyAccount render snapshot 1`] = `
Array [
  <Styled(View)>
    <Header
      onBackClick={[Function]}
    />
    <Styled(KeyboardAwareFlatList)
      ListFooterComponent={[Function]}
      ListHeaderComponent={[Function]}
      data={
        Array [
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.changeNotification",
            "value": 0,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.accountHacked",
            "value": 1,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.cantFindInterestingJobs",
            "value": 2,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.madeAnotherAccount",
            "value": 3,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.dontLikeWebsite",
            "value": 4,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.privacyConcern",
            "value": 5,
          },
          Object {
            "onClick": [Function],
            "text": "deleteAccountScreen.other",
            "value": 6,
          },
        ]
      }
      extraData={
        Array [
          "",
          false,
          2,
        ]
      }
      keyExtractor={[Function]}
      renderItem={[Function]}
    />
    <Styled(View)>
      <Button
        linear={true}
        loading={false}
        onClick={[Function]}
        testID="submitButton"
      >
        DELETEACCOUNTSCREEN.DELETEMYACCOUNT
      </Button>
    </Styled(View)>
  </Styled(View)>,
  <ForwardRef>
    <InputBlock
      animatedLine={true}
      disabled={true}
      label="deleteAccountScreen.writeYourReason"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="deleteAccountScreen.typeHere"
      value=""
    />
  </ForwardRef>,
  <ForwardRef>
    <ForwardRef>
      deleteAccountScreen.deleteYourAccount
    </ForwardRef>
    <ForwardRef>
      deleteAccountScreen.whyYouAreLeaving
      .
    </ForwardRef>
  </ForwardRef>,
  <ForwardRef>
    <UNDEFINED
      disabled={false}
      isActive={false}
      onClick={[Function]}
    />
    <ForwardRef>
      text
      .
    </ForwardRef>
  </ForwardRef>,
]
`;
